sphinx/source/conf.py

```diff
@@ -275,3 +270,70 @@
     'pandas': ('http://pandas.pydata.org/pandas-docs/stable/', None),
     'numpy': ('https://docs.scipy.org/doc/numpy/', None)
 }
+
+
+import os
+import subprocess
+
+# Do not use Bokeh CDNs to serve JS files when building latest version
+# of docs (and/or unreleased versions). We want to serve the compiled
+# JS files from Read the Docs servers itself to match the
+# documentation with the JS version. Although, when using a tagged
+# release, we want to use Bokeh CDN.
+def _get_current_commit():
+    result = subprocess.run(
+        ['git', 'rev-parse', '--verify', 'HEAD'],
+        stdout=subprocess.PIPE,
+    )
+    return result.stdout.decode('utf8').strip()
+
+def is_tagged_version():
+    commit = _get_current_commit()
+    print(f'Bokeh commit: {commit}')
+    result = subprocess.run(
+        ['git', 'tag', '--points-at', commit],
+        stdout=subprocess.PIPE,
+    )
+    tagged = result.stdout.decode('utf8').strip()
+    print(f'Bokeh tagged version: {tagged}')
+    return bool(tagged)
+
+if os.environ.get('READTHEDOCS') == 'True':
+    readthedocs_version = os.environ.get('READTHEDOCS_VERSION')
+    if is_tagged_version():
+        # Use Bokeh Resources from CDN
+        os.environ['BOKEH_DOCS_CDN'] = readthedocs_version
+    else:
+        # Use Bokeh Resources from Read the Docs
+        os.environ['BOKEH_DOCS_CDN'] = f'test:{readthedocs_version}'
+
+    bokeh_docs_cdn = os.environ.get('BOKEH_DOCS_CDN')
+    print(f'BOKEH_DOCS_CDN: {bokeh_docs_cdn}')
```

Comment:
We should avoid f-strings for now. We still support Python 2.7 until Bokeh 2.0 and even after that we will still support Python 3.5 and we should make it so anyone can contribute to the docs using a supported version